vitosmt.cc

```diff
@@ -128,20 +125,16 @@ vitosmt_read()
 
     auto* wpt_tmp = new Waypoint;
 
-    wpt_tmp->latitude =DEG(latrad);
-    wpt_tmp->longitude =DEG(lonrad);
-    wpt_tmp->altitude =elev;
+    wpt_tmp->latitude = DEG(latrad);
+    wpt_tmp->longitude = DEG(lonrad);
+    wpt_tmp->altitude = elev;
 
-    tmStruct.tm_year =timestamp[0]+100;
-    tmStruct.tm_mon =timestamp[1]-1;
-    tmStruct.tm_mday =timestamp[2];
-    tmStruct.tm_hour =timestamp[3];
-    tmStruct.tm_min =timestamp[4];
-    tmStruct.tm_sec  =(int)floor(seconds);
-    tmStruct.tm_isdst =-1;
+    QDate wdate(timestamp[0] + 2000, timestamp[1], timestamp[2]);
 
     double usec = fmod(1000000*seconds+0.5,1000000);
```

Comment:
This is problematic.  It can result in incorrect rounding, e.g.
seconds: 59.9999999 usec: 0.3999999985098839